### DISCUSSION TITLE
Unable to add MacOS devices in meshcentral server

### DISCUSSION THREAD

**Discussion by bc210422179ash** [Ylianst/MeshCentral#5941] [Issues]
I have setup my meshcentral server 1.1.21 in aws in wanonly mode, its working fine with windows and linux clients, but i'm unable to add any MacOS device as below mentioned process, i have tried both macos agent and apple osx binary, when i run agent and install no mac devices shows on meshcentral server, niether mac requires any permission after installing agent.

Server Software 
OS: Amazon linux 2
Version: 1.1.21
Node: 16.20.2

Client Device (please complete the following information):
Device: MacBook Air 2020 M1 processor, inteel
OS: MacOS Ventura 13.3,

**Comment by si458** [comment]
Yes the is a few bugs with the mac meshagent!
You need to install using the binary files, then make a few changes to get it to start correctly
1. Install using binary (download universal then chmod then -install as sudo)
2. Then you need to change the /Library/LaunchAgents/meshagent.plist Label to say meshagent-launchagent
3. Then restart mac
4. Then you to go into the web ui, into the device, then console and agentupdate (this is important! So it downloads the arm version instead)
5. Then restart mac, then try connecting from the web ui and it shouod show prompts, you might need to disconnect/reconnect a few times to get it to show correctly
